### PR TITLE
add spire server

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

### CODE REVIEW

**Comment by dlorenc** [comment]
Nice!

**Comment by dlorenc** [comment]
This is amazing :)

File: spire-server.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - busybox
+      - go
+      - build-base
+      - git
+pipeline:
+  - uses: fetch
+    with:
+      uri: https://github.com/spiffe/spire/archive/v${{package.version}}/v${{package.version}}.tar.gz
+      expected-sha256: 08a8f97771afae82024b3b407ac4dafe284cd8658ac02855141607b6508b6da6
+  - runs: |
```
Comment (imjasonh) [inline spire-server.yaml:28]: Maybe this should be a subpackage, so it's clearer they're related, and we only have to bump the version for both in one place.

Comment (developer-guy) [reply spire-server.yaml:28]: I tried to make it work with sub-packages, but I couldn't. Would you mind helping me to figure this out @imjasonh @tuananh? What am I doing wrong?

File: spire-server.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      make bin/spire-agent
+      make bin/spire-server
+      install -Dm755 ./bin/spire-server "${{targets.destdir}}/usr/bin/spire-server"
+      install -Dm755 ./bin/spire-agent "${{targets.destdir}}/usr/bin/spire-agent"
+  - uses: strip
+subpackages:
+  - name: spire-agent
+    description: The SPIFFE Runtime Environment (SPIRE) agent
+    pipelines:
```
Comment (rawlingsj) [inline spire-server.yaml:37]: this needs to be `pipeline:`

File: spire-server.yaml (hunk 1 above)
Comment (rawlingsj) [reply spire-server.yaml:28]: Created an issue to track adding a validation lint rule https://github.com/wolfi-dev/wolfictl/issues/17

File: spire-server.yaml (hunk 2 above)
Comment (developer-guy) [reply spire-server.yaml:37]: Yep, it worked that way 🤦🏻‍♂️ thx so much. Can you please take a look at my other PR in melange?

https://github.com/chainguard-dev/melange/pull/238

File: spire-server.yaml (hunk 1 above)
Comment (developer-guy) [reply spire-server.yaml:28]: that's a perfect issue for @dentrax !

**Review by rawlingsj — DISMISSED** [review]

**Review by dlorenc — DISMISSED** [review]

**Comment by dlorenc** [comment]
Needs a rebsse

**Review by rawlingsj — APPROVED** [review]

**Review by dlorenc — APPROVED** [review]

**Pull request merged**